examples/07-EMIT/interference_gui.py

```diff
@@ -0,0 +1,509 @@
+
+        # Create list of problem types to analyze according to inputted filters
+        filter = [i for (i,v) in zip(self.interference_filters, self.interference_checks) if v]
+
+        if self.file_path_box.text() != "" and self.design_name_dropdown.currentText() != "":
+            if self.previous_design != self.design_name_dropdown.currentText() or self.previous_project != self.file_path_box.text():
+                # self.emitapp.close_project()
+                self.previous_design = self.design_name_dropdown.currentText()
+                self.previous_project = self.file_path_box.text()
+                self.emitapp.set_active_design(self.design_name_dropdown.currentText())
+            
+                self.rev = self.emitapp.results.analyze()
+                self.modeRx = TxRxMode().RX
+                self.modeTx = TxRxMode().TX
+                self.mode_power = ResultType().POWER_AT_RX
+                self.tx_interferer = InterfererType().TRANSMITTERS
+
+                self.rx_radios = self.rev.get_receiver_names()
+                self.tx_radios = self.rev.get_interferer_names(self.tx_interferer)
+                self.domain = self.emitapp.results.interaction_domain()
+                self.radios = self.emitapp.modeler.components.get_radios()
+
+                if self.tx_radios is None or self.rx_radios is None:
+                    return
+
+            # Initialize results arrays
+            self.power_matrix=[]
+            self.all_colors=[]
+            
+            for tx_radio in self.tx_radios:
+                rx_powers = []
+                rx_colors = []
+                for rx_radio in self.rx_radios:
+                    # powerAtRx is the same for all Rx bands, so just use first one
+                    rx_bands = self.rev.get_band_names(rx_radio, self.modeRx)
+                    rx_band_objects = self.radios[rx_radio].bands()
+                    if tx_radio == rx_radio:
+                        # skip self-interaction
+                        rx_powers.append('N/A')
+                        rx_colors.append('white')
+                        continue       
+
+                    max_power = -200
+                    tx_bands = self.rev.get_band_names(tx_radio, self.modeTx)
+                    tx_band_objects = self.radios[tx_radio].bands()
+
+                    for i in range(len(rx_bands)):
+                        
+                        rx_freq = self.rev.get_active_frequencies(rx_radio, rx_bands[i], self.modeRx)
+                        rx_start_freq = self.radios[rx_radio].band_start_frequency(rx_band_objects[i])
+                        rx_stop_freq = consts.unit_converter(float(rx_band_objects[i].props["StopFrequency"]), "Freq", "Hz", "MHz")
+                        rx_channel_bandwidth = consts.unit_converter(float(rx_band_objects[i].props["ChannelBandwidth"]), "Freq", "Hz", "MHz")
+                        
+                        for j in range(len(tx_bands)):
```

Comment:
Are these the same loops as ComputeInterferenceType or are they computing something else? If they're computing the same values, can we delete this and replace with calls to functions within ComputeInterferenceType? Or make a third file that contains common code that both ComputeInterferenceType (the scripted version?) and this file can leverage so code isn't duplicated.

Reply:
The loops are very similar, I created a third file (interference_classification.py) which both the gui and the script make function calls to now.

Reply:
The separate file looks good. Do you want to update ComputeProtectionLevels.py to use your new function call too?